File: ios/Classes/model/ActiveGeofenceWires.swift
import CoreLocation

class ActiveGeofenceWires {
    static func fromGeofence(_ geofence: Geofence) -> ActiveGeofence {
        let currentTimeMillis = Int64(Date().timeIntervalSince1970 * 1000)
        return ActiveGeofence(
            id: geofence.id,
            location: geofence.location,
            radiusMeters: geofence.radiusMeters,
            triggers: geofence.triggers,
            androidSettings: geofence.androidSettings,
            status: .active,
            createdAtMillis: currentTimeMillis,
            statusChangedAtMillis: currentTimeMillis
        )
    }

    static func fromRegion(_ region: CLRegion) -> ActiveGeofence? {
        guard let circularRegion = region as? CLCircularRegion else { return nil }
        
        // Normalize coordinates from iOS CLRegion for consistency
        let normalizedLocation = Location(
            latitude: circularRegion.center.latitude,
            longitude: circularRegion.center.longitude
        ).normalized()
        
        let currentTimeMillis = Int64(Date().timeIntervalSince1970 * 1000)
        return ActiveGeofence(
            id: circularRegion.identifier,
            location: normalizedLocation,
            radiusMeters: circularRegion.radius,
            triggers: [
                circularRegion.notifyOnEntry ? .enter : nil,
                circularRegion.notifyOnExit ? .exit : nil
            ].compactMap { $0 },
            androidSettings: nil,
            status: .active,
            createdAtMillis: currentTimeMillis,
            statusChangedAtMillis: currentTimeMillis
        )
    }
}

File: android/src/main/kotlin/com/steffaanus/native_geofence/util/ActiveGeofenceWires.kt
package com.steffaanus.native_geofence.util

import com.steffaanus.native_geofence.generated.ActiveGeofence
import com.steffaanus.native_geofence.generated.AndroidGeofenceSettings
import com.steffaanus.native_geofence.generated.Geofence
import com.steffaanus.native_geofence.generated.GeofenceStatus
import com.steffaanus.native_geofence.generated.Location
import com.google.android.gms.location.Geofence

class ActiveGeofenceWires {
    companion object {
        fun fromGeofence(e: Geofence): ActiveGeofence {
            return ActiveGeofence(
                e.id,
                e.location,
                e.radiusMeters,
                e.triggers,
                e.androidSettings,
                GeofenceStatus.ACTIVE
            )
        }

        fun fromGeofence(e: Geofence, status: GeofenceStatus): ActiveGeofence {
            return ActiveGeofence(
                e.id,
                e.location,
                e.radiusMeters,
                e.triggers,
                e.androidSettings,
                status,
            )
        }
    }
}
